File: java/src/main/java/DidKey.java
import com.google.protobuf.InvalidProtocolBufferException;
import trinsic.okapi.keys.v1.Keys;

public class DidKey extends OkapiNative {
    public static Keys.GenerateKeyResponse generate(Keys.GenerateKeyRequest request) throws DidException, InvalidProtocolBufferException {
        OkapiByteBuffer.ByValue requestBuffer = messageToBuffer(request);
        OkapiByteBuffer responseBuffer = new OkapiByteBuffer();
        ExternError errBuffer = new ExternError();
        getNativeLibrary().didkey_generate(requestBuffer, responseBuffer, errBuffer);
        errBuffer.RaiseError();
        return Keys.GenerateKeyResponse.parseFrom(bufferToByteArray(responseBuffer));
    }

    public static Keys.ResolveResponse resolve(Keys.ResolveRequest request) throws DidException, InvalidProtocolBufferException {
        OkapiByteBuffer.ByValue requestBuffer = messageToBuffer(request);
        OkapiByteBuffer responseBuffer = new OkapiByteBuffer();
        ExternError errBuffer = new ExternError();
        getNativeLibrary().didkey_resolve(requestBuffer, responseBuffer, errBuffer);
        errBuffer.RaiseError();
        return Keys.ResolveResponse.parseFrom(bufferToByteArray(responseBuffer));
    }
}

File: ruby/lib/did_key.rb
# frozen_string_literal: true

# DidKey module
module DidKey
  def self.generate(request)
    Okapi.verify_type(request, Keys_V1::GenerateKeyRequest)
    Okapi.ffi_call('didkey_generate', request, Keys_V1::GenerateKeyResponse)
  end

  def self.resolve(request)
    Okapi.verify_type(request, Keys_V1::ResolveRequest)
    Okapi.ffi_call('didkey_resolve', request, Keys_V1::ResolveResponse)
  end
end
